cluster-provision/gocli/go.mod

```diff
@@ -1,65 +1,172 @@
+	github.com/inconshreveable/mousetrap v1.1.0 // indirect
+	github.com/klauspost/compress v1.17.8 // indirect
+	github.com/moby/patternmatcher v0.6.0 // indirect
 	github.com/moby/sys/sequential v0.5.0 // indirect
-	github.com/moby/term v0.0.0-20201216013528-df9cb8a40635 // indirect
-	github.com/morikuni/aec v1.0.0 // indirect
 	github.com/opencontainers/go-digest v1.0.0 // indirect
-	github.com/opencontainers/image-spec v1.1.0-rc2.0.20221005185240-3a7f492d3f1b // indirect
+	github.com/opencontainers/image-spec v1.1.0 // indirect
 	golang.org/x/sys v0.21.0 // indirect
 	golang.org/x/term v0.21.0 // indirect
 	golang.org/x/text v0.16.0 // indirect
 	golang.org/x/tools v0.21.1-0.20240508182429-e35e4ccd0d2d // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect; gindirect
 )
+
+replace (
+	github.com/kubernetes-incubator/external-storage => github.com/libopenstorage/external-storage v0.20.4-openstorage-rc3
+	github.com/libopenstorage/secrets => github.com/rook/secrets v0.0.0-20240315053144-3195f6906937
+	github.com/opencontainers/runc => github.com/opencontainers/runc v1.1.1-0.20240131200429-02120488a4c0
+	github.com/openshift/api => github.com/openshift/api v0.0.0-20230406152840-ce21e3fe5da2
+	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20230324103026-3f1513df25e0
+	github.com/openshift/library-go => github.com/mhenriks/library-go v0.0.0-20230310153733-63d38b55bd5a
+	github.com/operator-framework/operator-lifecycle-manager => github.com/operator-framework/operator-lifecycle-manager v0.0.0-20190128024246-5eb7ae5bdb7a
+	github.com/portworx/sched-ops => github.com/portworx/sched-ops v0.20.4-openstorage-rc3
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.28.4
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.28.4
+	k8s.io/client-go => k8s.io/client-go v0.30.1
+	k8s.io/cloud-provider => k8s.io/cloud-provider v0.28.4
+	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.28.4
+	k8s.io/component-base => k8s.io/component-base v0.28.4
+	k8s.io/cri-api => k8s.io/cri-api v0.28.4
+	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.28.4
+	k8s.io/endpointslice => k8s.io/staging/src/k8s.io/endpointslice v0.28.4
+	k8s.io/klog => k8s.io/klog v0.4.0
+	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.28.4
+	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.28.4
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.28.4
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.28.4
+	k8s.io/kubectl => k8s.io/kubectl v0.28.4
+	k8s.io/kubelet => k8s.io/kubelet v0.28.4
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.28.4
+	k8s.io/metrics => k8s.io/metrics v0.28.4
+	k8s.io/node-api => k8s.io/node-api v0.28.4
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.28.4
+	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.28.4
+	k8s.io/sample-controller => k8s.io/sample-controller v0.28.4
+	kubevirt.io/qe-tools => kubevirt.io/qe-tools v0.1.8
+)
```

Comment:
these were used by some of the dependencies introduced, particularly the kubernetes dependencies that we import to get their schemas. prometheus, rook, etc..
without the replaces when you fetch those k8s packages they fail because they try to download invalid versions of their own dependencies